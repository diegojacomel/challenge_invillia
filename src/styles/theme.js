const primaryTheme = {
    colors: {
        primary: '#007bff',
        success: '#28a745',
        secondary: '#6c757d',
        warning: "#ffc107",
        black: '#000000',
        white: '#FFFFFF',
        gray: '#cccccc',
        grayLight: '#F2F2F2',
        grayDark: '#333333'
    },
    fontSize: {
        xs: "14px",
        sm: "16px",
        md: "20px",
        lg: "25px",
        hg: "40px",
        hhg: "60px"
    },
    spacing: {
        none: '0',
        xxs: '3px',
        xs: '5px',
        ss: '8px',
        sm: '10px',
        ssm: '15px',
        md: '20px',
        mmd: '25px',
        lg: '30px',
        llg: '35px',
        xg: '40px',
        xxg: '50px',
        gt: '80px',
        ggt: '100px',
        hg: '200px',
        hhg: '400px',
        quarter: '25%',
        middle: '50%',
        full: '100%'
    },
    doubleSpacing: {
        xs: "2px 5px",
        sm: "3px 10px",
        md: "5px 12px",
        lg: "8px 16px",
        xg: "10px 20px",
        hg: "20px 40px"
    },
    rounded: {
        none: '0',
        xs: '2px',
        sm: '3px',
        md: '5px',
        lg: '10px'
    }
}

export {
    primaryTheme
}