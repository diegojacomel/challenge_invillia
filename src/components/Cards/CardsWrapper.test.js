// Modules
import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components';

// Theme
import { primaryTheme } from '../../styles/theme';

// Component
import CardWrapper from './CardsWrapper';

describe('CardWrapper', () => {
    const wrapper = shallow(<CardWrapper>Content</CardWrapper>);

    test('Render without crashing', () => {
        shallow(<CardWrapper></CardWrapper>);
    })

    test('It exist', () => {
        expect(wrapper.find('#testCardWrapper')).toExist();
    })

    test('At least one', () => {
        expect(wrapper).toContainMatchingElement('#testCardWrapper');
    })

    test('Have display name', () => {
        expect(wrapper.find('#testCardWrapper')).toHaveDisplayName('CardWrapperElement');
    })

    test('Match selector', () => {
        expect(wrapper.find('#testCardWrapper')).toMatchSelector('CardWrapperElement');
    })

    it('Render correctly', () => {
        const tree = renderer.create(
            <ThemeProvider theme={primaryTheme}>
                <CardWrapper></CardWrapper>
            </ThemeProvider>
        ).toJSON();

        expect(tree).toMatchSnapshot();
    })
})