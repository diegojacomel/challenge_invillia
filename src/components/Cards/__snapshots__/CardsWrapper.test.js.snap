// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardWrapper Render correctly 1`] = `
<div
  className="sc-bdVaJa EpHdP"
/>
`;
